### DISCUSSION TITLE
Basic script for getting started quickly

### DISCUSSION THREAD

**Discussion by ThioJoe** [openai/whisper#1195] [Show and tell]
I wrote a ready-made script for doing basic transcriptions of video files. It's what I use for transcribing my own videos, so I figured others might find it useful if they aren't sure how to get started:

Gist here: https://gist.github.com/ThioJoe/e84c4e649857bf0a290b49a2caa670f7

It is pretty basic but has a few more features than the example code in the main readme:
- **No editing of code file required on each run:** It prompts you for the path of the file, so you can just drag the file into the command prompt then hit Enter (as opposed to editing the code file to add the file path every time). Though if you want to change the model or language you'll have to go into the code to change those variables.
- **Automatic naming of output files:** It will automatically create a directory called "Output" then put transcriptions into new text files based on the original video file name.
- **Pre-coded support for additional parameters:** If you want, you can change some more advanced parameters like `verbose` and `word_timestamps`. In addition to the transcription, you can choose whether to output a json file wit timestamps/segment data setting the `exportTimestampData` variable to True. This will also include word-level timestamps if that option is enabled.
   - However keep in mind the json file will definitely need to be further processed to be useful, as it contains a bunch of other data outputted by Whisper.

**Comment by notrydo** [comment]
Thanks!

**Comment by Frosti7** [comment]
Can you please add support for Diarization and speaker identification?

**Reply by ThioJoe** [reply]
No, I don't know how

**Reply by phineas-pta** [reply]
it's more complicated, but someone already did it:
 - colab version: https://github.com/ANonEntity/WhisperWithVAD
 - local commandline version: https://gitlab.com/s-tier-plots/vad-whisper-ai
